shared/p2p/negotiation.go

```diff
@@ -0,0 +1,76 @@
+package p2p
+
+import (
+	"context"
+
+	ggio "github.com/gogo/protobuf/io"
+	iconnmgr "github.com/libp2p/go-libp2p-interface-connmgr"
+	inet "github.com/libp2p/go-libp2p-net"
+	peer "github.com/libp2p/go-libp2p-peer"
+	pb "github.com/prysmaticlabs/prysm/proto/beacon/p2p/v1"
+)
+
+const handshakeProtocol = prysmProtocolPrefix + "/handshake"
+
+var _ = iconnmgr.ConnManager(&negotiator{})
+
+type negotiator struct {
+	contractAddress string
+}
+
+// Notifee ...
+func (n *negotiator) Notifee() inet.Notifiee {
+	return &inet.NotifyBundle{
+		ConnectedF: func(net inet.Network, conn inet.Conn) {
+			go func() {
```

Comment:
Without this goroutine, this method is blocking forever while waiting for a response on that protocol, but the peer's protocol handler is never hit.

With this goroutine, it seems the stream is immediately reset and/or some junk data is manifested from somewhere. The error is that the protobuf cannot be unmarshalled